input.cpp

```diff
@@ -12,80 +12,95 @@
 
 // functions that read from input files
 namespace marxan {
+    void to_lower(std::string& str)
+    {
+        transform(str.begin(), str.end(), str.begin(), ::tolower);
+    }
 
-    string storeFieldName(const vector<string>& varlist, int numvars, string sVarName,
-        const vector<string>& head, string fname) {
-
-        // add a field name from an input file header to a list
-        int foundit = 0;
-        int temp, newlink;
-
-        for (int i = 0; (i < numvars && foundit == 0); i++)
-        {
-            if (varlist[i].compare(sVarName) == 0)
-                foundit++;
-        }
-
-        if (!head.empty() && find(head.begin(), head.end(), sVarName) != head.end())
+    vector<string> get_tokens(const std::string& str)
+    {
+        static const string delimeters(" ,;:^*\"/\t\'\\\n");
+        vector<string> tokens;
+        string word;
+        for (char ch : str)
         {
-            displayErrorMessage("ERROR: variable %s has been defined twice in data file %s.\n", sVarName.c_str(), fname.c_str());
+            if (delimeters.find_first_of(ch) == string::npos)
+                word.push_back(ch);
+            else
+            {
+                tokens.push_back(word);
+                word.clear();
+            }
         }
-
-        return(sVarName);
+        if(!word.empty())
+            tokens.push_back(word);
+        return tokens;
     }
 
-    vector<string> GetFieldNames(string readname, string fname, FILE*& fp, int& ivars, const vector<string>& varList) {
-        char sLine[500];
+
+    vector<string> GetFieldNames(string readname, string fname, ifstream&  fp, const vector<string>& varList) 

```

Comment:
See comment above, this function too can be deleted and replaced with a much simpler function. I believe its intention was to just read a line and return the list of header names